category/category_repository_blackbox_test.go

```diff
@@ -0,0 +1,206 @@
+package category_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/almighty/almighty-core/category"
+	errs "github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/gormsupport/cleaner"
+	"github.com/almighty/almighty-core/gormtestsupport"
+	"github.com/almighty/almighty-core/migration"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type categoryRepoBlackBoxTest struct {
+	gormtestsupport.DBTestSuite
+	repo  category.Repository
+	clean func()
+	ctx   context.Context
+}
+
+func TestRunCategoryRepoBlackBoxTest(t *testing.T) {
+	suite.Run(t, &categoryRepoBlackBoxTest{DBTestSuite: gormtestsupport.NewDBTestSuite("../config.yaml")})
+}
+
+func (test *categoryRepoBlackBoxTest) SetupSuite() {
+	test.DBTestSuite.SetupSuite()
+	ctx := migration.NewMigrationContext(context.Background())
+	test.DBTestSuite.PopulateDBTestSuite(ctx)
+}
+
+func (test *categoryRepoBlackBoxTest) SetupTest() {
+	test.repo = category.NewRepository(test.DB)
+	test.clean = cleaner.DeleteCreatedEntities(test.DB)
+}
+
+func (test *categoryRepoBlackBoxTest) TearDownTest() {
+	test.clean()
+}
+
+// TestCreateLoadValidCategory tests create and load valid category
+func (test *categoryRepoBlackBoxTest) TestCreateLoadValidCategory() {
+	category1 := category.Category{
+		Name: "Backlog",
+	}
+
+	test.T().Run("create and load (valid)", func(t *testing.T) {
+		result1, err := test.repo.Create(test.ctx, &category1) // Create
+		require.Nil(test.T(), err)
+		require.NotNil(test.T(), result1)
+		require.NotNil(test.T(), result1.ID)
+		require.NotNil(test.T(), result1.Name)
```

Comment:
Hm, you're writing new tests here and you could easily use golden files here. But if you don't feel like doing it now. That's fine by me.